tests/run-make/extern-fn-reachable/rmake.rs

```diff
@@ -0,0 +1,10 @@
+//@ ignore-cross-compile
+use run_make_support::object::ObjectSymbol;
+use run_make_support::rustc;
+use run_make_support::symbols::assert_contains_exact_symbols;
+fn main() {
+    rustc().input("dylib.rs").output("dylib.so").prefer_dynamic().run();
+    assert_contains_exact_symbols("dylib.so", &["fun1", "fun2", "fun3", "fun4", "fun5"], |sym| {
+        dbg!(dbg!(sym).is_global()) && !dbg!(sym.is_undefined())
+    });
```

Comment:
Remark: original Makefile only tested for contains not exact match, I think one of them being the exact naming is different.

Reply:
hmm....that still feels pretty fragile to me, maybe we should just add an underscore on macOS

any idea what's going on with the symbol types?

Reply:
Can you clarify what you mean by "symbol types"?

Reply:
Suggestion: see above, macOS expects dylib symbols to be prefixed with `_`.